sdk/src/asset_handlers/jpeg_io.rs

```diff
@@ -63,13 +63,46 @@ fn extract_xmp(seg: &JpegSegment) -> Option<String> {
 fn xmp_from_bytes(asset_bytes: &[u8]) -> Option<String> {
     if let Ok(jpeg) = Jpeg::from_bytes(Bytes::copy_from_slice(asset_bytes)) {
         let segs = jpeg.segments_by_marker(markers::APP1);
-        let xmp: String = segs.filter_map(extract_xmp).collect();
-        Some(xmp)
+        for seg in segs {
+            if let Some(xmp) = extract_xmp(seg) {
+                return Some(xmp);
+            }
+        }
+        None
     } else {
         None
     }
 }
 
+fn write_xmp(
+    input_stream: &mut dyn CAIRead,
+    output_stream: &mut dyn CAIReadWrite,
+    xmp: Option<&str>,
+) -> Result<()> {
+    let mut buf = Vec::new();
+    // read the whole asset
+    input_stream.rewind()?;
+    input_stream.read_to_end(&mut buf).map_err(Error::IoError)?;
+    let mut jpeg = Jpeg::from_bytes(buf.into()).map_err(|_err| Error::EmbeddingError)?;
+
+    let segments = jpeg.segments_mut();
+    segments.retain(|seg| {
+        !(seg.marker() == markers::APP1 && seg.contents().starts_with(XMP_SIGNATURE))
+    });
+
+    if let Some(xmp) = xmp {
```

Comment:
Do we always want to just replace the XMP or should we merge with existing values?

Reply:
Jumping in to a very old question, I would push for merging with existing values, given that many publishers will use both C2PA and IPTC (and Exif and other metadata) simultaneously to ensure maximum compatibility.